src/sourceItemWrapper.js

```diff
@@ -64,10 +62,12 @@ class SourceItemWrapper extends ItemWrapper {
             );
             Wikicite.setExtraField(this.item, 'citation', jsonCitations);
             this.saveHandler();
-        } else if (SAVE_TO === 'note') {
+        } else if (this._storage === 'note') {
             let note = Wikicite.getCitationsNote(this.item);
-            if (!citations.length && note) {
-                note.eraseTx();
+            if (!citations.length) {
+                if (note) {
```

Comment:
Is there any reason you split the if statement here?

Reply:
Yes, this makes sure that we always return from the function early (in line 71) if there are no citations. Previously, if there were no citations, but also no existing note, the function would continue running and create a new citations note which would contain just an empty array.

I guess this was never an issue before but with the new migration feature it would create empty citation notes for all items in the library without citations if you migrate from extra field to note storage.

Reply:
Oops, I missed the return moving inside. Gotcha!